### ISSUE TITLE
Unable to have no-unused-keys rule working

### ISSUE THREAD

**Issue by hchauvat** [intlify/eslint-plugin-vue-i18n#329]
### Summary
I'm not able to have the `no-unused-keys`  rule working

**Environment**

- **ESLint version:** 8.16.0
- **eslint-plugin-vue version:** 6.2.2
- **eslint-plugin-vue-i18n version:** 2.0.0
- **Node version:** 16.14.2

**Full configuration:**
```json
{
  "root": true,
  "env": {
    "browser": true,
    "node": true
  },
  "extends": ["eslint:recommended", "plugin:@intlify/vue-i18n/recommended"],
  "rules": {
    "@intlify/vue-i18n/no-unused-keys": [
      "warn",
      {
        "extensions": [".js", ".vue"]
      }
    ]
  },
  "parserOptions": {
    "parser": "@babel/eslint-parser"
  },
  "settings": {
    "vue-i18n": {
      "localeDir": "src/i18n/*.json",
      "messageSyntaxVersion": "^8.24.0"
    }
  }
}

```

**What did you do?**
en.json
```json
{
  "title": "Hello",
  "unusedKey": "not used one"
}
```
then
```shell
npx eslint --ext .js,.vue,.json src
```

**What did you expect to happen?**
I expect eslint to raise an `no-unused-keys` error on "unusedKey"

**What actually happened?**
No error was raised

**Repository to reproduce this issue**
[GitHub repo](https://github.com/hchauvat/eslint-plugin-vue-i18n-issue)

**Comment by ota-meshi** [comment]
Thank you for this issue.

You need to add the following configuration to .eslintrc.json.

```json

  "overrides": [
      {
          "files": ["*.json"],
          "parser": "jsonc-eslint-parser"
      }
  ]
```

**Comment by hchauvat** [comment]
@ota-meshi Thanks for your answers. On my real repo, not the one I created to reproduce the issue, I use a `.eslintrc.js` file, not a `.json` one. What should I put for the parser?
